analyzer/src/main/kotlin/managers/SpdxTransitiveDocument.kt

```diff
@@ -0,0 +1,329 @@
+    private fun unresolvedPackage(identifier: String, issues: MutableList<OrtIssue>): SpdxPackage? {
+        val issue = issueForPackage(identifier) ?: createAndLogIssue(
+            source = managerName,
+            message = "'$identifier' could neither be resolved to a 'package' nor to an 'externalDocumentRef'."
+        )
+
+        issues += issue
+        return null
+    }
+
+    /**
+     * Return an [OrtIssue] to be assigned to a package with the given [identifier]. Issues can occur when resolving
+     * external document references. This function checks whether the package [identifier] includes such a reference.
+     * If so, an issue recorded for this reference is returned.
+     */
+    private fun issueForPackage(identifier: String): OrtIssue? =
+        issuesByReferenceId[identifier.substringBefore(':', "")]
+}
+
+/**
+ * Resolve all external references to SPDX documents contained in [document], and recursively in all referenced
+ * documents. Use [loader] to load documents. Resolve relative URLs against [baseUri]. Store all encountered references
+ * and the documents they point to in [references]. Store issues encountered when resolving references in [issues] with
+ * [managerName] as source of the issues. Use [knownUris] to detect cycles.
+ */
+private fun resolveTransitiveReferences(
+    loader: SpdxDocumentLoader,
+    managerName: String,
+    document: SpdxDocument,
+    baseUri: URI,
+    references: MutableMap<SpdxExternalDocumentReference, SpdxDocument>,
+    issues: MutableMap<String, OrtIssue>,
+    knownUris: MutableSet<URI>
+) {
+    document.resolveReferences(loader, baseUri, managerName).forEach { (ref, resolvedDoc) ->
+        resolvedDoc.document?.let { document ->
+            references += ref to document
+            if (knownUris.add(resolvedDoc.uri)) {
+                resolveTransitiveReferences(
+                    loader,
+                    managerName,
+                    document,
+                    resolvedDoc.uri,
+                    references,
+                    issues,
+                    knownUris
+                )
+            }
+        }
+
+        resolvedDoc.issue?.let { issues += ref.externalDocumentId to it }
+    }
+}
+
+/**
+ * Return a map with all [SpdxPackage]s found in one of the given [references] using qualified identifiers as keys.
+ */
+private fun collectPackages(
+    references: MutableMap<SpdxExternalDocumentReference, SpdxDocument>
+): Map<String, SpdxPackage> {
+    val allPackages = mutableMapOf<String, SpdxPackage>()
+
+    references.forEach { (reference, document) ->
+        allPackages += document.getPackages("${reference.externalDocumentId}:")
+    }
+
+    return allPackages
+}
+
+/**
+ * A data class to hold the result of an operation to resolve an [SpdxDocument] from an external reference. Resolving
+ * of the document may fail, then the document is *null*, and a corresponding [OrtIssue] is present.
+ */
+private data class DocumentResolveResult(
```

Comment:
Maybe "DocumentResolutionResult" or just "ResolutionResult"?

Reply:
Renamed to `Resolutionresult`.